### PR TITLE
add option for when filesystem still lives in the experimental namespace

### PR DESCRIPTION
**Pull request by mithodin**
For gcc-7, filesystem still lives in the experimental namespace, thus compilation will fail. I have added an option to meson to fix the header includes and namespaces accordingly.

### CODE REVIEW

**Comment by Alexays** [comment]
I'm looking for if there's no better option than a meson option, because it's not very friendly to the end user

**Comment by mithodin** [comment]
I've looked around a bit for some kind of auto-detection method, but I've not been able to find something. Possibly one could write a small test file and see whether it compiles or not.

**Comment by mithodin** [comment]
I've figured it out :-)

File: meson.build [hunk 1]
```diff
@@ -83,7 +83,11 @@ if dbusmenu_gtk.found()
     )
 endif
 
-if get_option('filesystem-experimental')
+test_fs = '''#include <filesystem>
+int main(void){}'''
+compiler = meson.get_compiler('cpp')
+
+if not compiler.compiles(test_fs, name : 'is filesystem still in experimental?')
```
Comment (Alexays) [inline meson.build]: if not compiler.has_header('filesystem')

Comment (mithodin) [reply meson.build]: nice, that is even simpler. I've updated the pr.

**Comment by colemickens** [comment]
Thanks for this fix, I now don't have to patch Waybar to package it for Nix: https://github.com/colemickens/nixpkgs-wayland/commit/4a8d7a5ccd6104366a8d5663040af6352ff580be !

**Comment by mithodin** [comment]
@colemickens Yup, part of why I created this pr is to avoid having to write a patch for my gentoo ebuild ^^

**Pull request merged**